src/main/java/org/ow2/proactive/catalog/dto/CatalogObjectMetadata.java

```diff
@@ -141,17 +143,21 @@ public CatalogObjectMetadata(String bucketName, String name, String projectName,
         if (projectName != null && !projectName.isEmpty()) {
             this.projectName = projectName;
         } else {
-            this.projectName = getProjectNameIfExistsOrEmptyString();
+            this.projectName = getMetadataValueIfExistsOrEmptyString("project_name");
+        }
+        if (tags != null && !tags.isEmpty()) {
+            this.tags = tags;
+        } else {
+            this.tags = getMetadataValueIfExistsOrEmptyString("tags");
         }
         this.extension = extension;
     }
 
-    private String getProjectNameIfExistsOrEmptyString() {
-        Optional<Metadata> projectNameIfExists = metadataList.stream()
-                                                             .filter(property -> property.getKey()
-                                                                                         .equals("project_name"))
-                                                             .findAny();
-        return projectNameIfExists.map(value -> value.getValue() == null ? "" : value.getValue()).orElse("");
+    private String getMetadataValueIfExistsOrEmptyString(String metadataKey) {
+        Optional<Metadata> metadataIfExists = metadataList.stream()
+                                                          .filter(property -> property.getKey().equals(metadataKey))
+                                                          .findAny();
+        return metadataIfExists.map(value -> value.getValue() == null ? "" : value.getValue()).orElse("");
```

Comment:
I think it's better to put the second test (`value.getValue() == null ?`) in the filter.
Also you can directly return.

Something like

```
return metadataList.stream()
                .filter(property -> property.getKey().equals(metadataKey) && !StringUtils.isBlank(property.getValue()))
                .findFirst()
                .map(Metadata::getValue)
                .orElse("");
```